packages/near-fast-auth-signer/src/utils/multi-chain/multiChain.ts

```diff
@@ -1,82 +1,59 @@
 import * as bitcoin from 'bitcoinjs-lib';
 import { BigNumberish } from 'ethers';
-import { Account } from 'near-api-js';
 
 import { Bitcoin } from './chains/Bitcoin';
 import EVM from './chains/EVM';
-import { ChainSignatureContracts } from './signature';
-
-type BaseTransaction = {
-  to: string;
-  value: string;
-  derivedPath: string
-};
-
-type EVMTransaction = BaseTransaction
-
-type BTCTransaction = BaseTransaction
-
-type ChainProviders = {
-  providerUrl: string,
-  contract: ChainSignatureContracts
-}
-
-type EVMChainConfig = {
-  type: 'EVM',
-}
-
-type BTCChainConfig = {
-  type: 'BTC'
-  networkType: 'bitcoin' | 'testnet'
-}
-
-type ChainConfig = EVMChainConfig | BTCChainConfig
+import {
+  BTCChainConfigWithProviders,
+  BTCTransaction,
+  ChainSignatureContracts,
+  EVMChainConfigWithProviders,
+  EVMTransaction,
+  Request,
+  Response,
+} from './chains/types';
+
+export const signAndSendEVMTransaction = async (
+  req: Request
+): Promise<Response> => {
+  try {
+    const evm = new EVM({
+      ...req.chainConfig,
+      relayerUrl: req.fastAuthRelayerUrl,
+    });
 
-type EVMChainConfigWithProviders = ChainProviders & EVMChainConfig
-type BTCChainConfigWithProviders = ChainProviders & BTCChainConfig
+    const { hash } = await evm.handleTransaction(
+      req.transaction as EVMTransaction,
+      req.nearAuthentication,
+      req.transaction.derivedPath
+    );
 
-type Request = {
-  transaction: EVMTransaction | BTCTransaction;
-  chainConfig: EVMChainConfigWithProviders | BTCChainConfigWithProviders;
-  account: Account;
-  fastAuthRelayerUrl: string;
+    return {
+      transactionHash: hash,
+      success:         true,
+    };
+  } catch (e) {
+    return {
+      success:      false,
+      errorMessage: e.message,
+    };
+  }
 };
 
-type SuccessResponse = {
-  transactionHash: string;
-  success: true;
-}
-
-type FailureResponse = {
-  success: false;
-  errorMessage: string;
-}
-
-type Response = SuccessResponse | FailureResponse
-
-const signAndSend = async (req: Request): Promise<Response> => {
+export const signAndSendBTCTransaction = async (
+  req: Request
+): Promise<Response> => {
   try {
-    let txid: string;
-
-    if (req.chainConfig.type === 'EVM') {
-      const evm = new EVM({ ...req.chainConfig, relayerUrl: req.fastAuthRelayerUrl });
-
-      txid = (await evm.handleTransaction(
-        req.transaction,
-        req.account,
-        req.transaction.derivedPath,
-      )).hash;
-    } else if (req.chainConfig.type === 'BTC') {
-      const btc = new Bitcoin({ ...req.chainConfig, relayerUrl: req.fastAuthRelayerUrl });
+    const btc = new Bitcoin({
+      ...(req.chainConfig as BTCChainConfigWithProviders),
```

Comment:
Same here, we should not use casting inside chain specific methods. The Request args should be BTC specific